### DISCUSSION TITLE
delete lines completely

### DISCUSSION THREAD

**Discussion by ousia** [orbitalquark/textadept#211] [Q&A] [answered]
Thanks to @rgieseke in #210, I can delete a complete line with:

```lua
keys['ctrl+k'] = function() 
  buffer.del_line_right()
  buffer.del_line_left()
end
```

But now I realize that I need to remove also the final character for a new line (regexp `\n`).

And I also need that if I select more than a line, I can remove all selected lines. Right now, only the last line is removed.

Which is the right way to do this?

Many thanks for your help.

**Comment by clayton256** [comment, marked as answer]
If I understand correctly what you’re looking for, I did it with:

keys['cmd+d'] = function()
    buffer.line_delete()
end

**Reply by ousia** [reply]
This was partly what I intended, but my explanation was crappy.

```lua
keys['ctrl+k'] = function( )
  buffer.clear()
  buffer.line_delete()
end
```

I wanted to completely remove multiple selected lines.

This may be scary code, but it does the trick.

Many thanks for your help.
